legacy/src/main/java/org/opensearch/sql/legacy/plugin/RestSqlAction.java

```diff
@@ -162,7 +162,7 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
             return channel -> executeSqlRequest(request, queryAction, client, channel);
         } catch (Exception e) {
             logAndPublishMetrics(e);
-            return channel -> reportError(channel, e, isClientError(e) ? BAD_REQUEST : SERVICE_UNAVAILABLE);
+            return channel -> reportError(channel, e, isClientError(e) ? BAD_REQUEST : SERVICE_UNAVAILABLE, newSqlQueryHandler.getErrorStr());
```

Comment:
isn't this just an empty string?

Reply:
To enter this catch statement, 2 exceptions must be thrown. The first being the new SQL engine, second being the legacy. When new engine exception is caught this will not be an empty string, but the error message returned from query execution.

Reply:
KK - thanks